pkg/api/types.go

```diff
@@ -180,6 +180,139 @@ type VolumeSource struct {
 	GitRepo *GitRepoVolumeSource `json:"gitRepo"`
 	// Secret represents a secret that should populate this volume.
 	Secret *SecretVolumeSource `json:"secret"`
+	// PersistentVolumeClaimVolumeSource represents a reference to a PersistentVolumeClaim in the same namespace
+	PersistentVolumeClaimVolumeSource *PersistentVolumeClaimVolumeSource `json:"persistentVolumeClaim,omitempty"`
+}
+
+// Similar to VolumeSource but meant for the administrator who creates PVs.
+type PersistentVolumeSource struct {
+
+	// GCEPersistentDisk represents a GCE Disk resource that is attached to a
+	// kubelet's host machine and then exposed to the pod.
+	GCEPersistentDisk *GCEPersistentDiskVolumeSource `json:"persistentDisk"`
+
+	// AWSElasticBlockStore represents an EBS volume in AWS that is attached to
+	// kubelet's host machine and then exposed to the pod.
+	AWSElasticBlockStore *AWSElasticBlockStore `json:"elasticBlockStore"`
+
+	// NFSMount represents the location of an exported volume on a fileserver
+	NFSMount *NFSMount `json:"nfsMount"`
+
+	// HostPath represents a directory on the host.  It is useful for testing PersistentVolumes.
+	HostPath *HostPathVolumeSource `json:"hostPath"`
```

Comment:
Except it's not so persistent.  I'm concerned mostly because even I started to say "if you just add a hostname here then you have local data persistent storgae - you just need to make the scheduler binding smarter" - slippery slope.   If we leave this in, I want it very clear that it does not imply that data will be copied across machines - just for testing